packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -156,60 +114,58 @@ export const useShippingMethod = props => {
     /*
      *  Effects.
      */
+    // Issue the query only if we have a cartId.
     useEffect(() => {
         if (cartId) {
-            fetchShippingMethods({
+            fetchShippingMethodInfo({
                 variables: { cartId }
             });
         }
-    }, [cartId, fetchShippingMethods]);
-
-    useEffect(() => {
-        if (cartId) {
-            fetchSelectedShippingMethod({
-                variables: { cartId }
-            });
-        }
-    }, [cartId, fetchSelectedShippingMethod]);
-
-    useEffect(() => {
-        try {
-            const chosenShippingMethod =
-                chosenShippingMethodData.cart.shipping_addresses[0]
-                    .selected_shipping_method;
-
-            const nextDisplayState = chosenShippingMethod
-                ? displayStates.DONE
-                : displayStates.EDITING;
-
-            setDisplayState(nextDisplayState);
-        } catch {
-            setDisplayState(displayStates.EDITING);
-        }
-    }, [chosenShippingMethodData]);
-
-    // Simple heuristic to check shipping data existed prior to this render.
-    // On first submission, when we have data, we should tell the checkout page
-    // so that we set the next step correctly.
-    const hasData =
-        chosenShippingMethodData &&
-        chosenShippingMethodData.cart.shipping_addresses.length &&
-        chosenShippingMethodData.cart.shipping_addresses[0]
-            .selected_shipping_method;
+    }, [cartId, fetchShippingMethodInfo]);
 
+    // When we have data we should tell the checkout page
+    // so that it can set the step correctly.
```

Comment:
I wanted to bring up this scenario but I'm not sure if it's in scope here. If you change your shipping address and are required to re-select a shipping method then the form displays both the `Continue to Payment Information` button as well as the Payment Info section and Review Order button.

[![Image from Gyazo](https://i.gyazo.com/048bf077b463745769a191dab9eb8ab3.gif)](https://gyazo.com/048bf077b463745769a191dab9eb8ab3)

Reply:
Great call, I forgot to mention that. I was hoping we could create a separate issue for this particular "edge case" of going back.